src/Fmt_ast.ml

```diff
@@ -2048,12 +2048,12 @@ and fmt_class_structure c ~ctx ~parens ?ext self_ fields =
            $ ( match fields with
              | ({pcf_desc= Pcf_attribute a}, _) :: _
                when Option.is_some (fst (doc_atrs [a])) ->
-                 fmt "\n"
+                 fmt (eol_fmt ())
```

Comment:
Given the changes in Translation_unit that set the formatter output function, are these changes in Fmt_ast strictly needed? Can you explain what goes wrong without them? These changes are much more invasive than just those in Translation_unit, and make the code harder to change, and harder to keep right as it's non-obvious to know when \n is ok and when it isn't. It might even be better to just post-process the output with `tr` or whatever to get the desired line endings.

Reply:
Yes these changes are needed, otherwise there are many '\n' remaining in the output when line-endings = windows. The only '\n' we have to keep are those in literal strings (`"a\nb"`).
If I can get the same results with post-processing the output I will do it.